src_c/circle.c

```diff
@@ -334,21 +333,168 @@ pg_circle_collideswith(pgCircleObject *self, PyObject *arg)
     else if (PySequence_Check(arg)) {
         double x, y;
         if (!pg_TwoDoublesFromObj(arg, &x, &y)) {
-            return RAISE(
+            PyErr_SetString(
                 PyExc_TypeError,
                 "Invalid point argument, must be a sequence of two numbers");
+            return -1;
         }
         result = pgCollision_CirclePoint(scirc, x, y);
     }
     else {
-        return RAISE(PyExc_TypeError,
-                     "Invalid shape argument, must be a Circle, Rect / FRect, "
-                     "Line, Polygon or a sequence of two numbers");
+        PyErr_SetString(
+            PyExc_TypeError,
+            "Invalid point argument, must be a sequence of 2 numbers");
+        return -1;
+    }
+
+    return result;
+}
+
+static PyObject *
+pg_circle_collideswith(pgCircleObject *self, PyObject *arg)
+{
+    int result = _pg_circle_collideswith(&self->circle, arg);
+    if (result == -1) {
+        return NULL;
     }
 
     return PyBool_FromLong(result);
 }
 
+static PyObject *
+pg_circle_collidelist(pgCircleObject *self, PyObject *arg)
+{
+    Py_ssize_t i;
+    pgCircleBase *scirc = &self->circle;
+    int colliding;
+
+    if (!PySequence_Check(arg)) {
+        return RAISE(PyExc_TypeError, "Argument must be a sequence");
```

Comment:
Even though the method has only one argument, I think this error msg is a bit more clear and helpful:
```suggestion
        return RAISE(PyExc_TypeError, "colliders argument must be a sequence");
```